src/socketio.quiver.js

```diff
@@ -263,23 +300,59 @@ QuiverSocialProvider.prototype.login = function(loginOpts, continuation) {
             this.connectAsClient_.bind(this, friend, null)).
                 catch(onConnectionFailure.bind(this, friend)));
       }
-      Promise.all(connectionPromises).then(finishLogin);
+      return Promise.all(connectionPromises);
     }.bind(this);
 
     this.connectLoop_(this.configuration_.self.servers,
-        this.connectAsOwner_.bind(this)).then(connectToFriends, connectToFriends);
+        this.connectAsOwner_.bind(this)).then(function(results) {
```

Comment:
Fixed!  I simplified this to just checking whether there was an error or not.  No error => successful connection.